### PR TITLE
Add number of driftignore lines to telemetry

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #443
| ❓ Documentation  | yes 

## Description

This PR adds number of driftignore lines to telemetry.

### CODE REVIEW

**Review by lotoussa — DISMISSED** [review]

**Comment by wbeuil** [comment]
Can you put this one in draft for now since the #647 will probably change your `RulesCount()` method.

**Comment by sundowndev** [comment]
> Can you put this one in draft for now since the #647 will probably change your `RulesCount()` method?

Tested #647 but it does not seem to be ready, may be I can merge this one today and @moadibfr will just have few conflicts to solve

**Comment by wbeuil** [comment]
> > Can you put this one in draft for now since the #647 will probably change your `RulesCount()` method?
> 
> @wbeuil Tested #647 but it does not seem to be ready, may be I can merge this one today and @moadibfr will just have few conflicts to solve

We're gonna talk this one on our next sprint meeting and see what should be the next step.

**Review by lotoussa — APPROVED** [review]

**Review by wbeuil — DISMISSED** [review]
LGTM

File: pkg/analyser/analysis.go [hunk 1]
```diff
@@ -35,14 +35,15 @@ type Summary struct {
...
-	Date        time.Time
+	unmanaged        []resource.Resource
+	managed          []resource.Resource
+	deleted          []resource.Resource
+	differences      []Difference
+	summary          Summary
+	alerts           alerter.Alerts
+	Duration         time.Duration
+	Date             time.Time
+	IgnoreRulesCount int
```
Comment (eliecharra) [inline pkg/analyser/analysis.go:46]: I don't think this should be part of analysis struct

File: pkg/filter/driftignore.go [hunk 2]
```diff
@@ -51,18 +52,19 @@ func (r *DriftIgnore) readIgnoreFile() error {
...
 		line = strings.ReplaceAll(line, "/", separator)
 
-		lines = append(lines, gitignore.ParsePattern(line, nil))
+		patterns = append(patterns, gitignore.ParsePattern(line, nil))
 		if !strings.HasSuffix(line, "*") {
 			line := fmt.Sprintf("%s.*", line)
-			lines = append(lines, gitignore.ParsePattern(line, nil))
+			patterns = append(patterns, gitignore.ParsePattern(line, nil))
 		}
+		r.lines = append(r.lines, line)
```
Comment (eliecharra) [inline pkg/filter/driftignore.go:60]: Do we really need to store all the lines in memory ? Maybe a counter can be enough here like a `lineCount` or `rulesCount` as you named in in tests ?

File: pkg/telemetry/telemetry_test.go [hunk 3]
```diff
@@ -32,29 +32,33 @@ func TestSendTelemetry(t *testing.T) {
...
+				Duration:         123,
+				IgnoreRulesCount: 24,
 			},
 		},
 		{
 			name: "valid analysis with round up",
 			analysis: func() *analyser.Analysis {
 				a := &analyser.Analysis{}
 				a.Duration = 123.5 * 1e9 // 123.5 seconds
+				a.IgnoreRulesCount = 0
```
Comment (eliecharra) [inline pkg/telemetry/telemetry_test.go:53]: You can remove this I think as this is not the prupose of this test case. I'd rather prefer to have minimal test cas with a single responsibility

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: pkg/analyser/analysis.go (hunk 1 above)
Comment (sundowndev) [reply pkg/analyser/analysis.go:46]: Well the analysis object is what we use to transport every data about the scan, I'm afraid it's the best way to carry information along the execution

Comment (sundowndev) [reply pkg/analyser/analysis.go]: check my latest commit, I put ignoreRulesCount value in a dedicated return value

**Comment by sundowndev** [comment]
blocked by #705

**Comment by eliecharra** [comment]
Closing https://github.com/cloudskiff/driftctl/issues/443#issuecomment-880655915

**Pull request closed without merging**